added configuration options for high speed and speed clamp rails

porl opened the pull request:
Let me know if I am initialising it in a way that doesn't match your style.

Current options for:
- High Speed Rail
  - Max Speed
  - Acceleration if Occupied
  - Acceleration if Unoccupied
- Speed Clamp Rail
  - Clamp Speed

Haven't played with other blocks/items in the mod so don't know what should be configurable.

MrTroble commented on src/main/java/info/u_team/useful_railroads/config/Config.java:
```diff
@@ -0,0 +1,83 @@
+
+import java.nio.file.Path;
+
+public class Config {
```
I would suggest renaming this to prevent ambiguity

MrTroble commented on src/main/java/info/u_team/useful_railroads/config/Config.java:
```diff
@@ -0,0 +1,83 @@
+
+	@SubscribeEvent
+	public static void onLoad(final ModConfig.Loading configEvent) {
+
```
It seams like it's load and reload implementation are missing? I might be wrong is this intentional?

MrTroble commented:
Thank you for all the time you put into the development of this mod btw.

porl commented:
Thanks for the suggestions! I renamed Config and also removed those useless stubs. I've not worked on any modding since before the 1.7 days so I've been fumbling around on how things are done now. I used a basic config class example from a tutorial and obviously forgot to clear out the unused parts.

Happy to help out - these are changes we wanted on our little private server so I figured if you can use them it is better to commit upstream than have to maintain it myself!

HyCraftHD commented on src/main/java/info/u_team/useful_railroads/UsefulRailroadsMod.java:
```diff
@@ -10,6 +14,11 @@
+
+		ModLoadingContext.get().registerConfig(ModConfig.Type.COMMON, Config.COMMON_CONFIG);
+
+		Config.loadConfig(Config.COMMON_CONFIG, FMLPaths.CONFIGDIR.get().resolve("usefulrailroads-common.toml"));
```
Im not sure what 
```java
Config.loadConfig(Config.COMMON_CONFIG, FMLPaths.CONFIGDIR.get().resolve("usefulrailroads-common.toml")); 
```
should do. The config is loaded automatically if its registered

HyCraftHD commented on src/main/java/info/u_team/useful_railroads/config/Config.java:
```diff
@@ -0,0 +1,83 @@
+
+	public static void loadConfig(ForgeConfigSpec spec, Path path) {
+
+		final CommentedFileConfig configData = CommentedFileConfig.builder(path)
```
I see this here loads the config. Is this manual load required?
In other projects i just did this here:
https://github.com/MC-U-Team/Useful-Resources/blob/1.15.2/src/main/java/info/u_team/useful_resources/config/CommonConfig.java#L8-L27
and registered it here
https://github.com/MC-U-Team/Useful-Resources/blob/1.15.2/src/main/java/info/u_team/useful_resources/UsefulResourcesMod.java

porl commented:
I'm not sure actually, but just tested and I'm not sure if it is loading properly now. I will look at your code and see how to match things.

porl commented:
You were correct - the manual loading is not needed. Perhaps I was following an older style guide.

The configs load correctly for the client or from a server.

I noticed that setting both the high speed rails and the speed clamp rails to the same speed to not result in the same speed in game, so thought the config wasn't working but this appears to be unrelated - they both definitely update according to values in the config. As a separate pr I might look into why the way the high speed rail and the clamp rail sets velocity and see why.

HyCraftHD commented:
> I noticed that setting both the high speed rails and the speed clamp rails to the same speed to not result in the same speed in game, so thought the config wasn't working but this appears to be unrelated - they both definitely update according to values in the config. As a separate pr I might look into why the way the high speed rail and the clamp rail sets velocity and see why.

This is because the method speedUpCart which is used in the high speed rail uses this method to accelerate / teleport the minecart (see [here](https://pastebin.com/Dvk8xC9W)): https://github.com/MC-U-Team/Useful-Railroads/blob/aa4f623acd40976449cf47ee03922a1343cd962f/src/main/java/info/u_team/useful_railroads/block/CustomPoweredRailBlock.java#L114

The speed clamp rail sets only the motion: https://github.com/MC-U-Team/Useful-Railroads/blob/aa4f623acd40976449cf47ee03922a1343cd962f/src/main/java/info/u_team/useful_railroads/block/SpeedClampRailBlock.java#L31-L35
